src/components/color-picker/color-field/color-field.tsx

```diff
@@ -0,0 +1,437 @@
+    const canvasHeight = this.#canvasRef.height;
+    const canvasWidth = this.#canvasRef.width;
+
+    // Ensure valid pixel range
+    this.#currentX = Math.max(
+      0,
+      Math.min(canvasWidth, event.clientX - rect.left)
+    );
+    this.#currentY = Math.max(
+      0,
+      Math.min(canvasHeight, event.clientY - rect.top)
+    );
+
+    this.#colorPickedVariants = this.#getColorFromCanvas();
+    this.#drawMarker();
+    // Use debounced update for drag operations
+    this.#updateStatusMessage();
+  };
+
+  #handleCanvasClick = (event: MouseEvent) => {
+    this.#updateCoordinates(event);
+    // Immediate update for single clicks
+    this.#updateStatusMessage(true);
+    this.input.emit(this.#colorPickedVariants);
+  };
+
+  #handleMouseDown = (event: MouseEvent) => {
+    this.#isDragging = true;
+    this.#updateCoordinates(event);
+  };
+
+  #handleMouseMove = (event: MouseEvent) => {
+    if (this.#isDragging) {
+      this.#dragRAF.perform(
+        () => {
+          this.#updateCoordinates(event);
+        },
+        () => {
+          // Store coordinates immediately for responsiveness with proper clamping
+          const rect = this.#canvasRef.getBoundingClientRect();
+          const canvasHeight = this.#canvasRef.height;
+          const canvasWidth = this.#canvasRef.width;
+
+          this.#currentX = Math.max(
+            0,
+            Math.min(canvasWidth, event.clientX - rect.left)
+          );
+          this.#currentY = Math.max(
+            0,
+            Math.min(canvasHeight, event.clientY - rect.top)
+          );
+        }
+      );
+    }
+  };
+
+  #handleMouseUp = () => {
+    if (this.#isDragging) {
+      this.#isDragging = false;
+      this.input.emit(this.#colorPickedVariants);
+    }
+  };
+
+  #handleKeyDown = (event: KeyboardEvent) => {
+    const canvasHeight = this.#canvasRef?.height || 0;
+    const canvasWidth = this.#canvasRef?.width || 0;
+
+    switch (event.key) {
+      case KEY_CODES.ARROW_UP:
+        this.#currentY -= this.step;
+        if (this.#currentY < 0) {
+          this.#currentY = 0;
+        }
+        this.#drawMarker();
+        event.preventDefault();
+        break;
+      case KEY_CODES.ARROW_DOWN:
+        this.#currentY += this.step;
+        if (this.#currentY >= canvasHeight) {
+          this.#currentY = canvasHeight;
+        }
+        this.#drawMarker();
+        event.preventDefault();
+        break;
+      case KEY_CODES.ARROW_LEFT:
+        this.#currentX -= this.step;
+        if (this.#currentX < 0) {
+          this.#currentX = 0;
+        }
+        this.#drawMarker();
+        event.preventDefault();
+        break;
+      case KEY_CODES.ARROW_RIGHT:
+        this.#currentX += this.step;
+        if (this.#currentX >= canvasWidth) {
+          this.#currentX = canvasWidth;
+        }
+        this.#drawMarker();
+        event.preventDefault();
+        break;
+      default:
+        break;
+    }
```

Comment:
Use a dictionary to manage keyboard events.

https://github.com/genexuslabs/chameleon-controls-library/blob/main/src/components/tab/tab.tsx#L268